src/main/java/me/oyurimatheus/nossoservicodepagamento/payment/purchase/gateways/SaoriGateway.java

```diff
@@ -0,0 +1,48 @@
+package me.oyurimatheus.nossoservicodepagamento.payment.purchase.gateways;
+
+import me.oyurimatheus.nossoservicodepagamento.payment.purchase.Country;
+import me.oyurimatheus.nossoservicodepagamento.payment.purchase.Payment;
+import me.oyurimatheus.nossoservicodepagamento.payment.purchase.PaymentTransaction;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.net.URI;
+import java.util.EnumSet;
+import java.util.Optional;
+
+import static me.oyurimatheus.nossoservicodepagamento.payment.purchase.Country.BR;
+
+@Service
+class SaoriGateway implements PaymentGateway {
+
+    @Value("${nosso-servico-de-pagamentos.gateways.saori}")
+    private BigDecimal fare;
+    private final Country country = BR;
+    private final PaymentGatewayClient client;
+    private EnumSet<CreditCardFlag> cardFlags = EnumSet.allOf(CreditCardFlag.class);
+
+    SaoriGateway(PaymentGatewayClient client) {
+        this.client = client;
+    }
+
+    @Override
+    public Optional<PaymentTransaction> pay(Payment payment) {
+        boolean paymentSuccessful = client.pay(URI.create("http://localhost:8080/gateways/pay"), new PaymentGatewayClient.PaymentGatewayRequest(payment));
```

Comment:
Aqui você precisa se defender... Se o código for ficar assim, você precisa ter certeza que está pagando um Paymento que é accept por esse gateway :).

Reply:
Faz sentido! Nada impede que isso seja chamado em outro ponto do código em outro contexto e a gente precisa se defender disso também